metadata/metadata.go

```diff
@@ -824,14 +858,51 @@ func (this *modelResource) Notify(lookup ResourceLookup, op operation, that Reso
 }
 
 func (resource *modelResource) UpdateStatus(status pb.ResourceStatus) error {
-	resource.serialized.Status = &status
 	return nil
 }
 
 func (resource *modelResource) UpdateSchedule(schedule string) error {
 	return fmt.Errorf("not implemented")
 }
 
+func (resource *modelResource) Update(lookup ResourceLookup, updateRes Resource) error {
+	deserialized := updateRes.Proto()
+	updateModel, ok := deserialized.(*pb.Model)
+	if !ok {
+		return errors.New("failed to deserialize existing model record")
+	}
+	updated, err := unionNameVariants(resource.serialized.Features, updateModel.Features)
+	if err != nil {
+		return err
+	}
+	resource.serialized.Features = updated
+	updated, err = unionNameVariants(resource.serialized.Trainingsets, updateModel.Trainingsets)
+	if err != nil {
+		return err
+	}
+	resource.serialized.Trainingsets = updated
+	if err = lookup.Set(resource.ID(), resource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func unionNameVariants(source, destination []*pb.NameVariant) ([]*pb.NameVariant, error) {
+	set := make(map[string]bool)
+
+	for _, nameVariant := range destination {
+		set[fmt.Sprintf("%s|%s", nameVariant.Name, nameVariant.Variant)] = true
```

Comment:
FYI, @simba-git, after taking a closer look a the proto's `String` method, I noticed a comment about not relying on the output, which _could_ change from version to version:
<img width="1312" alt="Screen Shot 2023-02-14 at 9 19 25 AM" src="https://user-images.githubusercontent.com/10430775/218826790-683f6be4-1749-4c9b-9ce0-7263158a74b0.png">
I chose to handle the string formatting explicitly myself, but if there's a better pattern I can follow, let me know.

Reply:
I would make a struct and use that